### ISSUE TITLE
libvorbis contains a main function

### ISSUE THREAD

**Issue by 9chu** [amzeratul/halley#41]
`src/contrib/libvorbis/lib/barkmel.c` contains an `int main` entry that should be removed.

**Comment by gurka** [comment]
Better report this upstream, perhaps at https://github.com/xiph/vorbis ?

However, I don't think that this is an error. libvorbis contains three utility / extra programs: barkmel tone and psytune. So the corresponding c-files should have an `int main`.

**Comment by 9chu** [comment]
> Better report this upstream, perhaps at https://github.com/xiph/vorbis ?
> 
> However, I don't think that this is an error. libvorbis contains three utility / extra programs: barkmel, tone and psytune, see https://github.com/xiph/vorbis/blob/ea8b03fce93444cb3cf0131909e15b4f8856e863/lib/Makefile.am#L29 So the corresponding c-files should have an `int main`.

Yes, it's an utility program. So the real question is that the `CMakeLists.txt` is not supposed to include it.

In `src/engine/audio/CMakeLists.txt`:
```cmake
file (GLOB_RECURSE VORBIS_FILES "../../contrib/libvorbis/*.c")
```

That may lead to a link problem.

**Comment by gurka** [comment]
Aha, yes that is a problem. I wonder why they don't just add the libvorbis directory with `add_directory`. libvorbis uses CMake as well, so it should be possible.
